Libraries/PeriphDrivers/Include/MAX32520/smon.h

```diff
@@ -47,82 +47,127 @@ extern "C" {
  *
  */
 typedef enum {
-    SMON_EXTSENSOR_0 = MXC_F_SMON_EXTSCN_EXTS_EN0,
+    MXC_SMON_EXTSENSOR_0 = MXC_F_SMON_EXTSCN_EXTS_EN0,
+
+    // Deprecated Name
+    SMON_EXTSENSOR_0 = MXC_SMON_EXTSENSOR_0,
 } mxc_smon_ext_sensor_t;
 
 /**
  * @brief   Enum for clock divider
  *
  */
 typedef enum {
-    SMON_CLK_DIVIDE_1 = MXC_S_SMON_EXTSCN_DIVCLK_DIV1,
-    SMON_CLK_DIVIDE_2 = MXC_S_SMON_EXTSCN_DIVCLK_DIV2,
-    SMON_CLK_DIVIDE_4 = MXC_S_SMON_EXTSCN_DIVCLK_DIV4,
-    SMON_CLK_DIVIDE_8 = MXC_S_SMON_EXTSCN_DIVCLK_DIV8,
-    SMON_CLK_DIVIDE_16 = MXC_S_SMON_EXTSCN_DIVCLK_DIV16,
-    SMON_CLK_DIVIDE_32 = MXC_S_SMON_EXTSCN_DIVCLK_DIV32,
-    SMON_CLK_DIVIDE_64 = MXC_S_SMON_EXTSCN_DIVCLK_DIV64,
+    MXC_SMON_CLK_DIVIDE_1 = MXC_S_SMON_EXTSCN_DIVCLK_DIV1,
+    MXC_SMON_CLK_DIVIDE_2 = MXC_S_SMON_EXTSCN_DIVCLK_DIV2,
+    MXC_SMON_CLK_DIVIDE_4 = MXC_S_SMON_EXTSCN_DIVCLK_DIV4,
+    MXC_SMON_CLK_DIVIDE_8 = MXC_S_SMON_EXTSCN_DIVCLK_DIV8,
+    MXC_SMON_CLK_DIVIDE_16 = MXC_S_SMON_EXTSCN_DIVCLK_DIV16,
+    MXC_SMON_CLK_DIVIDE_32 = MXC_S_SMON_EXTSCN_DIVCLK_DIV32,
+    MXC_SMON_CLK_DIVIDE_64 = MXC_S_SMON_EXTSCN_DIVCLK_DIV64,
+
+    // Deprecated names
+    SMON_CLK_DIVIDE_1 = MXC_SMON_CLK_DIVIDE_1,
```

Comment:
You can deprecate enums with an attribute tag.  For example:

SMON_CLK_DIVIDE_1 __attribute__ ((deprecated ("replaced by MXC_SMON_CLK_DIVIDE_1"))) = MXC_SMON_CLK_DIVIDE_1,

Reply:
Wow, didn't know this. This is a game-changer!